gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -527,7 +531,7 @@ def prop_defined(name, cls, rules={})
     # *haven't* allowed additional options in the past and want to
     # default to keeping this interface narrow.
     @class.send(:define_method, fk_method) do |allow_direct_mutation: nil|
-      foreign = T.let(foreign, T.untyped)
+      foreign = T.let(foreign, T.untyped, checked: false)
```

Comment:
I think if you move this whole statement outside of the block, you will get the same benefit of type pinning in Sorbet, but you won't be making the `T.let` call on every call to `fk_method`.